### DISCUSSION TITLE
Trouble updating to new wrapper

### DISCUSSION THREAD

**Discussion by sood5cg** [The-Wineskin-Project/WineskinServer#204] [Q&A] [answered]
I have a Wineskin wrapper that worked perfectly under Mojave, and I'm trying to get it working on a new M1 Mac.  I've installed the latest Wineskin Winery and downloaded the latest wrapper (2.9.0.9-4) and some recent Crossover engines.

When I try to update my Wineskin (is that the right terminology?) to the latest wrapper, I get an error message – something about english.lproj, and the updated Wineskin doesn't run.  I saw something at https://github.com/Gcenx/WineskinServer/releases about incompatibility between some previous wrapper versions.  I assume it's the same issue here.  Specifically, "the previous master wrapper Wineskin-2.9.0.7-rc1 won't be able to update to this master wrapper version as the structure has changed."

I also see this immediately thereafter: "While it's technically possible to update a current wrapper to the rc2 I won't provide any support for this so please create the wrapper over and manually move over the wineprefix files currently located with the wrappers /Contents/Resources directory into the newly created wrapper.

I'm afraid I don't really understand those instructions, which means I'm asking for support, I guess.

I think a fresh install might work.  But the Wineskin contains a program (Vocaloid) with multiple voice banks that were all installed from their respective DVDs and individually activated.  It would be a big pain to have to do them all again.  So I'm hoping there's a way to manually update the Wineskin rather than installing everything from scratch in a fresh wrapper.

The previous wrapper version was 2.8.8.

Could you (or anyone) clarify the instructions I quoted above?  This is how I read them:
1. Create a new blank wrapper with the desired engine.
2. Move the wineprefix files from the old wrapper to the new one.  These should be in /Contents/Resources.

I have two problems with this.  First, I don't know how to identify a wineprefix file.  Do I move all the files in /Contents/Resources?  Second, at no point do the instructions say to move the actual program and its associated files to the new wrapper.  So I must be missing something.

Finally, lest I appear ungrateful by appearing out of nowhere and asking for help, let me say that Wineskin has been very helpful for me, and I'm very grateful to doh123 and Gcenx for bringing it to life and keeping it alive, respectively.

**Comment by sood5cg** [comment, marked as answer]
The following seems to resuscitate a wrapper that has failed its update process.  Copy the following four items from a new blank wrapper to the corresponding locations in the broken one: PkgInfo, SharedSupport, Base.lproj, and en.lproj.  Then choose the executable to run, if it's been changed incorrectly.

**Comment by Gcenx** [comment]
Firstly create a new blank wrapper with the desired name and Engine.

From the new wrapper remove the following files from `/Contents/Resources/`
- `dosdevices`
- `drive_c`
- `system.reg`
- `user.reg`
- `userdef.reg`

Now copy the same files from the old wrapper into the same location in the new wrapper that you just removed.\
Now launch `Wineskin.app` and select the target exe etc.


<br>

#### Please note
I've been slowly working on a new wrapper update that moves the prefix location _again_ but this will hopefully make things simpler in the future. The included wineprefix will be installed into `/Contents/SharedSupport/prefix` meaning it's even simpler to move the wrappers wineprefix around if/when needed.

**Reply by sood5cg** [reply]
Thanks for the reply.  I have no doubt your solution works, but I've tested mine and it's easier (imho) as well, so I'm marking mine as the answer.

**Reply by Gcenx** [reply]
> Thanks for the reply. I have no doubt your solution works, but I've tested mine and it's easier (imho) as well, so I'm marking mine as the answer.

While easier it’s not correct and won’t have all the files updated correctly and might cause additional problems.

After the next wrapper update I’ll lightly just disable “update wrapper” as it simply can’t function as intended.
